user_sync/config.py

```diff
@@ -60,83 +60,133 @@ def __init__(self, caller_options):
             'update_user_info': True,
             'username_filter_regex': None,
         }
-        options.update(caller_options)     
+        
+        # merge custom options with our default options
```

Comment:
okay sounds good, I'll try to refactoring the function descriptions and commenting